packages/flutter/lib/src/services/hardware_keyboard.dart

```diff
@@ -0,0 +1,894 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'binding.dart';
+import 'keyboard_key.dart';
+
+/// Represents a lock mode of a keyboard, such as [KeyboardLockMode.capsLock].
+///
+/// A lock mode locks part of a keyboard keys into a distinct mode of operation,
+/// depending on the lock settings selected. The status of the mode is toggled
+/// with each key down of its corresponding logical key. A [KeyboardLockMode]
+/// object is used to query whether this mode is enabled on the keyboard.
+///
+/// Only a limited number of modes are supported, which are enumerated as static
+/// members of this class. Manual constructing of this class is prohibited.
+@immutable
+class KeyboardLockMode {
+  // KeyboardLockMode has a fixed pool of supported keys, enumerated as static
+  // members of this class, therefore constructing is prohibited.
+  const KeyboardLockMode._(this.logicalKey);
+
+  /// The logical key that triggers this lock mode.
+  final LogicalKeyboardKey logicalKey;
+
+  /// Represents the number lock mode on the keyboard.
+  ///
+  /// On supporting systems, enabling number lock mode usually allows key
+  /// presses of the number pad to input numbers, instead of acting as up, down,
+  /// left, right, page up, end, etc.
+  static const KeyboardLockMode numLock = KeyboardLockMode._(LogicalKeyboardKey.numLock);
+
+  /// Represents the scrolling lock mode on the keyboard.
+  ///
+  /// On supporting systems and applications (such as spreadsheet), enabling
+  /// scrolling lock mode usually allows key presses of the cursor keys to
+  /// scroll the document instead of the cursor.
+  static const KeyboardLockMode scrollLock = KeyboardLockMode._(LogicalKeyboardKey.scrollLock);
+
+  /// Represents the capital lock mode on the keyboard.
+  ///
+  /// On supporting systems, enabling capital lock mode usually allows key
+  /// presses of the letter keys to input upper cases.
+  static const KeyboardLockMode capsLock = KeyboardLockMode._(LogicalKeyboardKey.capsLock);
+
+  static final Map<int, KeyboardLockMode> _knownLockModes = <int, KeyboardLockMode>{
+    numLock.logicalKey.keyId: numLock,
+    scrollLock.logicalKey.keyId: scrollLock,
+    capsLock.logicalKey.keyId: capsLock,
+  };
+
+  /// Returns the [KeyboardLockMode] constant from the logical key, or
+  /// null, if not found.
+  static KeyboardLockMode? findLockByLogicalKey(LogicalKeyboardKey logicalKey) => _knownLockModes[logicalKey.keyId];
+}
+
+/// Defines the interface for keyboard key events.
+///
+/// The [KeyEvent] provides a universal model for key event information from a
+/// hardware keyboard across platforms.
+///
+/// The [KeyEvent] is not to be created directly. Intead, use its subclasses
```

Comment:
It's abstract, so I don't think you need this paragraph: it's pretty clear you have to use a subclass if this is an abstract class.